Sources/Build/BuildPlan.swift

```diff
@@ -1344,6 +1344,19 @@ public final class ProductBuildDescription {
         }
     }
 
+    /// The arguments to the librarian to create a static library.
+    public func archiveArguments() throws -> [String] {
+        let librarian = buildParameters.toolchain.librarianPath.pathString
+        let triple = buildParameters.triple
+        if triple.isWindows(), librarian.hasSuffix("link") || librarian.hasSuffix("link.exe") {
+            return [librarian, "/LIB", "/OUT:\(binary.pathString)", "@\(linkFileListPath.pathString)"]
+        }
+        if triple.isDarwin(), librarian.hasSuffix("libtool") {
```

Comment:
Are these checks for the filename suffix temporary, or intended to be part of the final PR?  If the latter, I wonder how robust they are in case people substitute their own version that is intended to work the same as the official one.  Would it make sense to expect that whatever the librarian executable is called, it's expected to take a certain set of flags?  I think this is how the other tools are expected to work even if customized via an env var.

Reply:
They are intended to be the final version.  The reason is that the tools have differing command line invocations.  `lld-link` and `link` (which may be invoked with or without the `.exe` suffix) both use the MSVC style command line.  `libtool` and `llvm-libtool` would use the `libtool` like command line.  The other platforms use `ar` or `llvm-ar` and would use the Unix style command line.

There is a related issue on Windows.  The default tool should be `link`, and then if the user specifies to use `lld` (`-use-ld=lld`), we should prefer `lld-link`.  Finally, if the user sets `AR=llvm-ar` we should actually fallback to `ar` as the librarian.  On Darwin, `libtool` is the preferred librarian, though it _is_ possible to use `ar` (as was previously the case due to the use of the test rules in llbuild).

Reply:
I think the question is what happens if you supply an entirely custom tool. 

I wonder if we should just change the default? I think the `libtool`-style argument list makes more sense for a generic tool vs. the highly specific `ar` argument list.

Reply:
I think that `ar` style arguments make more sense - that is the default archiver on everything but Windows and Darwin.